specification/sql/resource-manager/Microsoft.Sql/preview/2021-11-01-preview/DatabaseAdvancedThreatProtection.json

```diff
@@ -0,0 +1,292 @@
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully retrieved the database Advanced Threat Protection states.",
+            "schema": {
+              "$ref": "#/definitions/DatabaseAdvancedThreatProtectionListResult"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 UpdateNotAllowedOnPausedDatabase - User attempted to perform an update on a paused database.\n\n * 400 AdvancedThreatProtectionInsufficientDiskSpace - Insufficient disk space to save the Advanced Threat Protection state metadata in the database\n\n * 400 InvalidDatabaseAdvancedThreatProtectionCreateRequest - The create database Advanced Threat Protection request does not exist or has no properties object\n\n * 400 AdvancedThreatProtectionInvalidUserSuppliedParameter - An invalid parameter value was provided by the client\n\n * 400 UpsertDatabaseAdvancedThreatProtectionFailed - An error has occurred while saving Advanced Threat Protection settings, please try again later\n\n * 400 UpsertDatabaseAdvancedThreatProtectionFailed - An error has occurred while saving Advanced Threat Protection settings, please try again later\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 DatabaseDoesNotExist - User has specified a database name that does not exist on this server instance.\n\n * 404 UpsertDatabaseAdvancedThreatProtectionFailed - An error has occurred while saving Advanced Threat Protection settings, please try again later\n\n * 409 DatabaseAdvancedThreatProtectionInProgress - Setting the database Advanced Threat Protection is already in progress\n\n * 409 UpsertDatabaseAdvancedThreatProtectionFailed - An error has occurred while saving Advanced Threat Protection settings, please try again later\n\n * 500 DatabaseIsUnavailable - Loading failed. Please try again later.\n\n * 500 UpsertDatabaseAdvancedThreatProtectionFailed - An error has occurred while saving Advanced Threat Protection settings, please try again later\n\n * 500 GetDatabaseAdvancedThreatProtectionFailed - Failed to get Advanced Threat Protection settings"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-examples": {
+          "Lists the database's Advanced Threat Protection states.": {
+            "$ref": "./examples/DatabaseAdvancedThreatProtectionListByDatabase.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Sql/servers/{serverName}/databases/{databaseName}/advancedThreatProtection/{advancedThreatProtectionName}": {
```

Comment:
>advancedThreatProtection [](http://example.com/codeflow?start=142&length=24)

plural advancedThreatProtections

Reply:
No, you cannot use that word in English in plural. It doesn't make sense. You don't have Protections in plural... it is a singular noun: protection (protection against all things).
Plus it is the name of the feature. That is the reason that I decided to ignore the warning that I got from the validation tools (I think it was from Avocado).

Reply:
A resourceType name is considered a collection of instances of that type so it needs to be plural as far as REST semantics are concerned. Please refer to the Azure API guidelines

[https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#:~:text=ex%3A%20blobcontainer%2C%20myqueue)-,resource%E2%80%91collection,-Name%20of%20the](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#:~:text=ex%3A%20blobcontainer%2C%20myqueue)-,resource%E2%80%91collection,-Name%20of%20the)

Reply:
consider advancedThreatProtectionPolicies or similar plural nouns